pkgs/build-support/setup-hooks/auto-patchelf.py

```diff
@@ -38,21 +40,42 @@ def is_dynamic_executable(elf: ELFFile) -> bool:
     return bool(elf.get_section_by_name(".interp"))
 
 
-def get_dependencies(elf: ELFFile) -> List[str]:
+def get_dependencies(elf: ELFFile) -> list[list[Path]]:
     dependencies = []
     # This convoluted code is here on purpose. For some reason, using
     # elf.get_section_by_name(".dynamic") does not always return an
     # instance of DynamicSection, but that is required to call iter_tags
     for section in elf.iter_sections():
         if isinstance(section, DynamicSection):
             for tag in section.iter_tags('DT_NEEDED'):
-                dependencies.append(tag.needed)
+                dependencies.append([Path(tag.needed)])
             break # There is only one dynamic section
 
     return dependencies
 
 
-def get_rpath(elf: ELFFile) -> List[str]:
+def get_dlopen_dependencies(elf: ELFFile) -> list[list[Path]]:
+    """
+    Extracts dependencies from the `.note.dlopen` section.
+    This is a FreeDesktop standard to annotate binaries with libraries that it may `dlopen`.
+    See https://systemd.io/ELF_DLOPEN_METADATA/
```

Comment:
I've seen that phrasing twice now, and so far all I could find was systemd documenting this and referring to the owner of the section header(?) as "FreeDesktop.org". Does anyone have any information on whether this is actually a FreeDesktop standard, because if I'm not mistaken systemd is neither under the freedesktop umbrella (they have moved elsewhere AFAIK), nor is any of this [listed in the FDO specs](https://www.freedesktop.org/wiki/Specifications/) (unless I just fail to see it right now).

This isn't a blocker or anything, I'm just curious. And if it happens to not be an actual FreeDesktop standard, could we please not call it that, and otherwise referring to the actual standard in question (rather than to a systemd related website) would be cool I guess?

Reply:
Hm, interesting. systemd calls it a FDO owned thing. In their documentation on their site, the documentation for the section header has this:

```
Owner: `FDO` (FreeDesktop.org)
```